### PR TITLE
rewrite API

### PR DESCRIPTION
**Pull request by rafalkanski**
@filipgolonka @wsromek @arkadiuszmalina @bartosz-kahl new approach 💃

### CODE REVIEW

File: src/Snt/Capi/Repository/Article/FindByIdsParameters.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * @param string $publicationId
+     * @param int[] $articleIds
+     */
+    public function __construct($publicationId, array $articleIds)
+    {
+        $this->publicationId = $publicationId;
+        $this->articleIds = $articleIds;
+    }
```
Comment (filipgolonka) [inline src/Snt/Capi/Repository/Article/FindByIdsParameters.php:25]: minor improvement - what do you think about introducing named constructor, i.e. `createForPublicationAndArticleIds` or even `createFrom` with array argument?

File: src/Snt/Capi/Repository/Article/FindParameters.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * @param string $publicationId
+     * @param int $articleId
+     */
+    public function __construct($publicationId, $articleId)
+    {
+        $this->publicationId = $publicationId;
+        $this->articleId = $articleId;
+    }
```
Comment (filipgolonka) [inline src/Snt/Capi/Repository/Article/FindParameters.php:25]: as above

Comment (filipgolonka) [inline src/Snt/Capi/Repository/Article/FindParameters.php:25]: moreover I think that those two classes can be merged into one, we can discuss it tomorrow

**Comment by filipgolonka** [comment]
:clap: :clap: :clap:

File: src/Snt/Capi/Repository/Article/FindByIdsParameters.php (hunk 1 above)
Comment (rafalkanski) [inline src/Snt/Capi/Repository/Article/FindByIdsParameters.php:25]: 👍

File: build.xml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project name="CAPI PHP SDK" default="all">
     <property name="package_src_directory"  value="src" />
+    <property name="package_features_directory"  value="features" />
```
Comment (arkadiuszmalina) [inline build.xml]: why 2 spaces before `value`? (the same 1 line above)

Comment (rafalkanski) [inline build.xml]: am i fired :( ?

**Comment by wsromek** [comment]
👍

**Comment by arkadiuszmalina** [comment]
👍

**Comment by filipgolonka** [comment]
:+1:

**Pull request merged**